platforms/ios/lib/WysiwygComposer/Sources/HTMLParser/Extensions/NSAttributedString+Range.swift

```diff
@@ -117,8 +139,16 @@ extension NSAttributedString {
     ///   - htmlIndex: the index inside the HTML raw text
     /// - Returns: the index inside the attributed representation
     func attributedPosition(at htmlIndex: Int) throws -> Int {
+        let discardableTextRanges = discardableTextRanges()
+        var actualIndex = htmlIndex
+
+        for discardableTextRange in discardableTextRanges {
+            if discardableTextRange.location < actualIndex {
+                actualIndex += discardableTextRange.length
+            }
+        }
```

Comment:
same here

but more generally I'm sure we can globally improve the code here, and perhaps merge list prefixes and discardable text into the same Array so we don't have multiple arrays that achieve same kind of computation

Reply:
Yeah not sure what happens in that case, technically on backspace we do replace the HTML so we should be re-parsing and adding that ZWSP.